presentation/src/main/java/com/anytypeio/anytype/presentation/sets/ViewerEditWidgetUi.kt

```diff
@@ -8,58 +8,35 @@ import com.anytypeio.anytype.core_models.ObjectWrapper
 import com.anytypeio.anytype.domain.objects.StoreOfObjectTypes
 import com.anytypeio.anytype.domain.objects.StoreOfRelations
 
-data class ViewerEditWidgetUi(
-    val showWidget: Boolean,
-    val id: Id? = null,
-    val name: String = "",
-    val viewerId: Id,
-    val defaultObjectType: ObjectWrapper.Type?,
-    val isDefaultObjectTypeEnabled: Boolean = false,
-    val layout: DVViewerType?,
-    val relations: List<Id> = emptyList(),
-    val filters: List<Id> = emptyList(),
-    val sorts: List<Id> = emptyList(),
-    val defaultTemplate: Id? = null,
-) {
 
-    fun empty() = this.copy(
-        id = null,
-        name = "",
-        layout = null,
-        relations = emptyList(),
-        filters = emptyList(),
-        sorts = emptyList(),
-        defaultTemplate = null,
-        defaultObjectType = null,
-        isDefaultObjectTypeEnabled = false,
-    )
-
-    companion object {
-        fun init() = ViewerEditWidgetUi(
-            showWidget = false,
-            id = null,
-            name = "",
-            layout = null,
-            relations = emptyList(),
-            filters = emptyList(),
-            sorts = emptyList(),
-            defaultTemplate = null,
-            defaultObjectType = null,
-            isDefaultObjectTypeEnabled = false,
-            viewerId = "",
-        )
-    }
+sealed class ViewerEditWidgetUi {
+    object Init: ViewerEditWidgetUi()
+    data class Data(
+        val showWidget: Boolean = false,
+        val showMore: Boolean = false,
+        val id: Id,
+        val name: String,
+        val defaultObjectType: ObjectWrapper.Type?,
+        val isDefaultObjectTypeEnabled: Boolean,
+        val layout: DVViewerType?,
+        val relations: List<Id> = emptyList(),
+        val filters: List<Id> = emptyList(),
+        val sorts: List<Id> = emptyList(),
+        val defaultTemplate: Id?
+    ) : ViewerEditWidgetUi()
+}
 
-    sealed class Action {
-        object Dismiss : Action()
-        data class UpdateName(val id: Id?, val name: String) : Action()
-        data class DefaultObjectType(val id: Id?) : Action()
-        data class Layout(val id: Id?) : Action()
-        data class Relations(val id: Id?) : Action()
-        data class Filters(val id: Id?) : Action()
-        data class Sorts(val id: Id?) : Action()
-        object More: Action()
-    }
+sealed class ViewEditAction {
+    object Dismiss : ViewEditAction()
+    data class UpdateName(val id: Id?, val name: String) : ViewEditAction()
+    data class DefaultObjectType(val id: Id?) : ViewEditAction()
+    data class Layout(val id: Id?) : ViewEditAction()
```

Comment:
all these data classes nullable ids mean that these values are optional?

Reply:
fixed. all ids are non-null